python/lsst/meas/extensions/trailedSources/NaivePlugin.py

```diff
@@ -282,6 +268,86 @@ def measure(self, measRecord, exposure):
+        y0: `float`
+            y coordinate of the beginning of the trail.
+        x1: `float`
+            x coordinate of the end of the trail.
+        y1: `float`
+            y coordinate of the end of the trail.
+        """
+        x_coords = [x0, x1]
+        y_coords = [y0, y1]
+
+        # Check if one of the end points of the trail sources is nan. If so,
+        # set the trailed source nan flag.
+        if np.isnan(x_coords).any() or np.isnan(y_coords).any():
+            self.flagHandler.setValue(measRecord, self.NAN.number, True)
+
+            x_coords = [x for x in x_coords if not np.isnan(x)]
+            y_coords = [y for y in y_coords if not np.isnan(y)]
+
+            # Check if the non-nan coordinates are within the bounding box
+            if not (all(exposure.getBBox().beginX <= x <= exposure.getBBox().endX for x in x_coords)
+                    and all(exposure.getBBox().beginY <= y <= exposure.getBBox().endY for y in y_coords)):
+
+                self.flagHandler.setValue(measRecord, self.EDGE.number, True)
+                self.flagHandler.setValue(measRecord, self.OFFIMAGE.number, True)
+
+            else:
+                # Check if edge is set for any of the pixel pairs. Do not
+                # check any that have a nan.
+
+                if x0 is not np.nan and y0 is not np.nan:
+                    if exposure.mask[Point2I(int(x0), int(y0))] & exposure.mask.getPlaneBitMask('EDGE') != 0:
+                        self.flagHandler.setValue(measRecord, self.EDGE.number, True)
+
+                if x1 is not np.nan and y1 is not np.nan:
+                    if exposure.mask[Point2I(int(x1), int(y1))] & exposure.mask.getPlaneBitMask('EDGE') != 0:
+                        self.flagHandler.setValue(measRecord, self.EDGE.number, True)
+
+        # Check whether trail extends off the edge of the exposure. Allows nans
+        # as their location
+        elif not (all(exposure.getBBox().beginX <= x <= exposure.getBBox().endX for x in x_coords)
+                  and all(exposure.getBBox().beginY <= y <= exposure.getBBox().endY for y in y_coords)):
+
+            self.flagHandler.setValue(measRecord, self.EDGE.number, True)
+            self.flagHandler.setValue(measRecord, self.OFFIMAGE.number, True)
+
+        else:
+            # Check whether the beginning or end point of the trail has the
+            # edge flag set. The end points are not whole pixel values, so
+            # the pixel value must be rounded.
+            if exposure.mask[Point2I(int(x0), int(y0))] and exposure.mask[Point2I(int(x1), int(y1))]:
+                if ((exposure.mask[Point2I(int(x0), int(y0))] & exposure.mask.getPlaneBitMask('EDGE') != 0)
+                        or (exposure.mask[Point2I(int(x1), int(y1))]
+                            & exposure.mask.getPlaneBitMask('EDGE') != 0)):
+                    self.flagHandler.setValue(measRecord, self.EDGE.number,
+                                              True)
```

Comment:
This dangling True feels exceptionally weirdly formatted. Perhaps run `black` after you run `isort` ...

Reply:
My auto formatter got the best of me, thanks for catching it!